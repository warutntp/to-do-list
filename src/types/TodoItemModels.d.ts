export module TodoItemModels {
  export interface TodoItemModel {
    slug: string;
    title: string;
    description?: string;
    completed: boolean;
  }
}
