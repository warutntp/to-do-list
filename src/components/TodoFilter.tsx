import React from "react";

interface FilterProps {
  currentFilter: string;
  setFilter: (filter: string) => void;
}

const TodoFilter: React.FC<FilterProps> = ({ currentFilter, setFilter }) => {
  const handleFilterChange = (filter: string) => () => {
    setFilter(filter);
  };

  return (
    <div className="btn-group">
      <button
        type="button"
        className={`btn ${
          currentFilter === "all" ? "btn-primary" : "btn-outline-primary"
        }`}
        onClick={handleFilterChange("all")}
      >
        All
      </button>
      <button
        type="button"
        className={`btn ${
          currentFilter === "incomplete" ? "btn-primary" : "btn-outline-primary"
        }`}
        onClick={handleFilterChange("incomplete")}
      >
        Incomplete
      </button>
      <button
        type="button"
        className={`btn ${
          currentFilter === "completed" ? "btn-primary" : "btn-outline-primary"
        }`}
        onClick={handleFilterChange("completed")}
      >
        Completed
      </button>
    </div>
  );
};

export default TodoFilter;
