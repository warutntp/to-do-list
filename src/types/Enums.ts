export enum FilterTodoType {
  Completed = "completed",
  Incomplete = "incomplete",
  All = "all",
}
