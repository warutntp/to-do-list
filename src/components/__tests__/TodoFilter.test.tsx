import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import TodoFilter from "../TodoFilter";
import { FilterTodoType } from "../../types/Enums";

describe("TodoFilter", () => {
  it("should match snapshot", () => {
    const setFilter = jest.fn();
    const { asFragment } = render(
      <TodoFilter currentFilter={FilterTodoType.All} setFilter={setFilter} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should call setFilter with the correct argument when a button is clicked", () => {
    const setFilter = jest.fn();
    render(
      <TodoFilter currentFilter={FilterTodoType.All} setFilter={setFilter} />
    );

    fireEvent.click(screen.getByText("Incomplete"));
    expect(setFilter).toHaveBeenCalledWith("incomplete");

    fireEvent.click(screen.getByText("Completed"));
    expect(setFilter).toHaveBeenCalledWith("completed");
  });
});
