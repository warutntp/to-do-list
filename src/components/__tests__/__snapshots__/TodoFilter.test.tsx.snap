// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFilter should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="btn-group"
  >
    <button
      class="btn btn-primary"
      type="button"
    >
      All
    </button>
    <button
      class="btn btn-outline-primary"
      type="button"
    >
      Incomplete
    </button>
    <button
      class="btn btn-outline-primary"
      type="button"
    >
      Completed
    </button>
  </div>
</DocumentFragment>
`;
